src/MapViewController.swift

```diff
@@ -13,57 +13,182 @@ import CoreLocation
+  /**
+   Asks the delegate if the map should recognize this single tap and perform default functionality.
+
+   - parameter controller: The MapViewController that wants to recognize the tap.
+   - parameter recognizer: The recognizer that initially recognized the tap.
+   - parameter location: The screen coordinates that the tap occured in relative to the bounds of the map.
+   
+   - returns: Return true for default functionality, or false if don't want it recognized (or plan on handling it yourself).
+  */
   func mapController(_ controller: MapViewController, recognizer: UIGestureRecognizer, shouldRecognizeSingleTapGesture location: CGPoint) -> Bool
+
+  /**
+   Informs the delegate the map recognized a single tap gesture.
+
+   - parameter controller: The MapViewController that recognized the tap.
+   - parameter recognizer: The recognizer that recognized the tap.
+   - parameter location: The screen coordinates that the tap occured in relative to the bounds of the map.
+   */
   func mapController(_ controller: MapViewController, recognizer: UIGestureRecognizer, didRecognizeSingleTapGesture location: CGPoint)
 }
 
+/// Double Tap Gesture Delegate
 public protocol MapDoubleTapGestureDelegate : class {
+  /**
+   Asks the delegate if the map should recognize this double tap and perform default functionality.
+
+   - parameter controller: The MapViewController that wants to recognize the tap.
+   - parameter recognizer: The recognizer that initially recognized the tap.
+   - parameter location: The screen coordinates that the tap occured in relative to the bounds of the map.
+
+   - returns: Return true for default functionality, or false if don't want it recognized (or plan on handling it yourself).
+   */
   func mapController(_ controller: MapViewController, recognizer: UIGestureRecognizer, shouldRecognizeDoubleTapGesture location: CGPoint) -> Bool
+
+  /**
+   Informs the delegate the map recognized a double tap gesture.
+
+   - parameter controller: The MapViewController that recognized the tap.
+   - parameter recognizer: The recognizer that recognized the tap.
+   - parameter location: The screen coordinates that the tap occured in relative to the bounds of the map.
+   */
   func mapController(_ controller: MapViewController, recognizer: UIGestureRecognizer, didRecognizeDoubleTapGesture location: CGPoint)
 }
 
+/// Long Press Gesture Delegate
 public protocol MapLongPressGestureDelegate : class {
+  /**
+   Asks the delegate if the map should recognize this long press gesture and perform default functionality.
+
+   - parameter controller: The MapViewController that wants to recognize the press.
+   - parameter recognizer: The recognizer that initially recognized the press.
+   - parameter location: The screen coordinates that the press occured in relative to the bounds of the map.
+
+   - returns: Return true for default functionality, or false if don't want it recognized (or plan on handling it yourself).
```

Comment:
Agreed, and I updated single-tap / double-tap / long-press with some more information. 

We're actually not exposing the shove / pinch / rotation gesture recognizers to third party devs at this point. We instead opted for a "enabled / not enabled" approach for those to simplify the front-facing API a bit, so we aren't documenting those here.